api/s2n.h

```diff
@@ -301,6 +301,9 @@ extern int s2n_config_add_ticket_crypto_key(struct s2n_config *config,
                                             const uint8_t *name, uint32_t name_len,
                                             uint8_t *key, uint32_t key_len,
                                             uint64_t intro_time_in_seconds_from_epoch);
+S2N_API
+extern int s2n_config_generate_local_ticket_crypto_key(struct s2n_config *config,
+                                            uint64_t intro_time_in_seconds_from_epoch);
```

Comment:
I like keeping intro_time_in_seconds_from_epoch to match intro_time_in_seconds_from_epoch. Maybe the names should match too? Like "s2n_config_add_random_ticket_crypto_key" or "s2n_config_add_generated_ticket_crypto_key" instead.

Reply:
good point. will consider keeping the word "add" for consistency and also to make it clear that this function modifies s2n_config, as @Matthias247 pointed out.

Reply:
Definite agreement with @lrstewart . I would expect `s2n_config_generate_local_ticket_crypto_key` to return the newly generated key rather than generating it and immediately associating it with the config. This is a very good method to give to our users, but lets make sure they can easily discover and use it.

Reply:
I waffled a bit on the name happy for more ideas. Settled on `s2n_config_generate_and_add_ticket_crypto_key`.